src/fw_algorithms.jl

```diff
@@ -88,7 +89,7 @@ function frank_wolfe(
     else
         similar(x)
     end
-    while t <= max_iteration && dual_gap >= max(epsilon, eps())
+    while t <= max_iteration && dual_gap >= max(epsilon, eps()) && stop_criterion != true
```

Comment:
I don't think we need an additional variable here. We can have a "break" statement if the result of the callback is true